### DISCUSSION TITLE
Are polyfills for core node modules required when using webpack 5?

### DISCUSSION THREAD

**Discussion by szekelyisz** [firebase/firebase-js-sdk#5519] [Q&A] [answered]
Hi, I'm trying to use Firebase Authentication in a Quasar Framework app. I'm importing the library as usual:

```typescript
import firebaseAuth from 'firebase/auth';
```

When I compile it, I get the following errors:

```
 App •  DONE  • "UI" compiled with errors • 10459ms



 App •  ERROR  •  UI  in ./node_modules/@google-cloud/firestore/build/src/bulk-writer.js

Module not found: Can't resolve imported dependency "assert"
Did you forget to install it? You can run: yarn add assert

 App •  ERROR  •  UI  in ./node_modules/@google-cloud/firestore/build/src/index.js

Module not found: Can't resolve imported dependency "stream"
Did you forget to install it? You can run: yarn add stream

 App •  ERROR  •  UI  in ./node_modules/@google-cloud/firestore/build/src/logger.js

Module not found: Can't resolve imported dependency "util"
Did you forget to install it? You can run: yarn add util

```

And the list goes on long, looks like something is trying to pull in built-in node modules. My guess is that this is cause by webpack 5 not providing automatic polyfills for code node modules anymore. There's [node-polyfill-webpack-plugin](https://github.com/Richienb/node-polyfill-webpack-plugin) as a workaround. Is this required for now or am I doing something wrong? Thanks!

**Comment by hsubox76** [comment, marked as answer]
I don't think Firebase has a `@google-cloud/firestore` dependency. Is there a special Firebase wrapper for Quasar that you are using? Also, the import looks unusual, `firebase/auth` shouldn't have a default export. Can you give more details about how you are using Firebase in this app?

**Reply by szekelyisz** [reply]
Thank you for your reply. You're right, I was trying to marry this repo with `@google-cloud/firestore` by mistake, that was trying to pull in Node modules. Then I realized it's only for Node and I actually need only this library.
